cmake/GzUtils.cmake

```diff
@@ -1944,6 +1944,159 @@ macro(gz_build_tests)
 
 endmacro()
 
+#################################################
+# gz_build_examples(
+#     SOURCE_DIR <source_dir>
+#     BINARY_DIR <binary_dir>
+#     [COMPONENTS <project_component_dependencies>]
+#     [DEPENDENCIES <project_dependencies>]
+#
+# Requires a CMakeLists.txt file to be in SOURCE_DIR that acts
+# Build examples for a Gazebo project. 
+# as a top level project.
+#
+# This generates two test targets
+# * EXAMPLES_Configure_TEST - Equivalent of calling "cmake .." on
+#                             the examples directory
+#
+# * EXAMPLES_Build_TEST - Equivalent of calling "make" on the
+#                         examples directory
+#
+# These tests are run during "make test" or can be run specifically
+# via "ctest -R EXAMPLES_ -V"
+#
+# Arguments are as follows:
+#
+# SOURCE_DIR: Required. Path to the examples folder.
+#             For example ${CMAKE_CURRENT_SOURCE_DIR}/examples
+#
+# BINARY_DIR: Required. Path to the output binary folder
+#             For example ${CMAKE_CURRENT_BINARY_DIR}/examples
+#
+# COMPONENTS: 
+#     Optional. List of components built by this project that are used in the examples.
+#     For example "cli" for gz-utils
+#
+# DEPENDENCIES: 
```

Comment:
Let me see if I understand this right: in `DEPENDENCIES` you can specify gz- libraries that are not being used by the build system of the project but are required to make the examples to build/run? Is this correct? If so, bellow there is a comment:

> we are passing in the resolved dependency directories via -Ddep_DIR=${dep_DIR}

How does this work if a dependency is not in the current buildsystem?

Reply:
I have a slight rework to this I can push shortly that drops this gymnastics a bit.  

Note that this is only necessary for gazebo packages, as they may be built in the same workspace, so this special forwarding happens.

Currently, examples aren't allowed to depend on Gazebo packages things that the containing package doesn't.  E.g. gz-gui examples cannot have a dependency on gz-gazebo, as this will introduce circular dependencies.

If an example depends on a non-Gazebo package, then typical `find_package` infrastructure should work.